test/dist_autograd_test.py

```diff
@@ -711,6 +713,21 @@ def test_backward_node_failure(self):
                 # Exit all other nodes.
                 pass
 
+    @dist_init(clean_shutdown=False)
+    def test_context_cleanup_timeout(self):
+        with dist_autograd.context() as context_id:
+            timeout = datetime.timedelta(seconds=2)
+            dist_autograd._setContextCleanupTimeout(timeout)
+            t1 = torch.rand((3, 3), requires_grad=True)
+            t2 = torch.rand((3, 3), requires_grad=True)
+
+            time.sleep(15)
```

Comment:
Shouldn't sleeping for `timeout * 2` be sufficient? Would be better to have this as a function of timeout. Also, if the watchdog sleeps for 60s, its not guaranteed that we would clean up the contexts within this timeout right?